`GivenDotnetPackageAdd` `noRestore=false` fails to work with package source mapping

A Package Add test is failing when no restore is set to false.
This appears to impact release/10.0.1xx and release/10.0.2xx at the very least.

`Microsoft.DotNet.Cli.Package.Add.Tests.GivenDotnetPackageAdd.Microsoft.DotNet.Cli.Package.Add.Tests.GivenDotnetPackageAdd.FileBasedApp_CentralPackageManagement_NoVersionSpecified_KeepExisting(legacyForm: True, fileOption: False, noRestore: False)`

```
Expected value to be 0 because Expected command to pass but it did not.
Working Directory: /datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5
File Name: /datadisks/disk1/work/AE91097D/p/d/dotnet
Arguments: add Program.cs package Humanizer
Exit Code: 1
StdOut:
Determining projects to restore...
/datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj : warning NU1507: There are 13 package sources defined in your configuration. When using central package management, please map your package sources with package source mapping (https://aka.ms/nuget-package-source-mapping) or specify a single package source. The following sources are defined: darc-pub-dotnet-dotnet-fad253f, dotnet6, dotnet7, dotnet8, dotnet8-transport, dotnet8-workloads, dotnet9, dotnet9-transport, dotnet10, dotnet10-transport, dotnet-public, dotnet-tools, test-tools
/datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj : error NU1101: Unable to find package Humanizer.Core.bn. No packages exist with this id in source(s): darc-pub-dotnet-dotnet-fad253f, dotnet-public, dotnet-tools, dotnet10, dotnet10-transport, dotnet6, dotnet7, dotnet8, dotnet8-transport, dotnet8-workloads, dotnet9, dotnet9-transport, Package source 1, Package source 2, test-tools
/datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj : error NU1101: Unable to find package Humanizer.Core.ca. No packages exist with this id in source(s): darc-pub-dotnet-dotnet-fad253f, dotnet-public, dotnet-tools, dotnet10, dotnet10-transport, dotnet6, dotnet7, dotnet8, dotnet8-transport, dotnet8-workloads, dotnet9, dotnet9-transport, Package source 1, Package source 2, test-tools
/datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj : error NU1101: Unable to find package Humanizer.Core.fi. No packages exist with this id in source(s): darc-pub-dotnet-dotnet-fad253f, dotnet-public, dotnet-tools, dotnet10, dotnet10-transport, dotnet6, dotnet7, dotnet8, dotnet8-transport, dotnet8-workloads, dotnet9, dotnet9-transport, Package source 1, Package source 2, test-tools
/datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj : error NU1101: Unable to find package Humanizer.Core.fil. No packages exist with this id in source(s): darc-pub-dotnet-dotnet-fad253f, dotnet-public, dotnet-tools, dotnet10, dotnet10-transport, dotnet6, dotnet7, dotnet8, dotnet8-transport, dotnet8-workloads, dotnet9, dotnet9-transport, Package source 1, Package source 2, test-tools
/datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj : error NU1101: Unable to find package Humanizer.Core.ko. No packages exist with this id in source(s): darc-pub-dotnet-dotnet-fad253f, dotnet-public, dotnet-tools, dotnet10, dotnet10-transport, dotnet6, dotnet7, dotnet8, dotnet8-transport, dotnet8-workloads, dotnet9, dotnet9-transport, Package source 1, Package source 2, test-tools
/datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj : error NU1101: Unable to find package Humanizer.Core.lb. No packages exist with this id in source(s): darc-pub-dotnet-dotnet-fad253f, dotnet-public, dotnet-tools, dotnet10, dotnet10-transport, dotnet6, dotnet7, dotnet8, dotnet8-transport, dotnet8-workloads, dotnet9, dotnet9-transport, Package source 1, Package source 2, test-tools
/datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj : error NU1101: Unable to find package Humanizer.Core.lt. No packages exist with this id in source(s): darc-pub-dotnet-dotnet-fad253f, dotnet-public, dotnet-tools, dotnet10, dotnet10-transport, dotnet6, dotnet7, dotnet8, dotnet8-transport, dotnet8-workloads, dotnet9, dotnet9-transport, Package source 1, Package source 2, test-tools
/datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj : error NU1101: Unable to find package Humanizer.Core.ms. No packages exist with this id in source(s): darc-pub-dotnet-dotnet-fad253f, dotnet-public, dotnet-tools, dotnet10, dotnet10-transport, dotnet6, dotnet7, dotnet8, dotnet8-transport, dotnet8-workloads, dotnet9, dotnet9-transport, Package source 1, Package source 2, test-tools
/datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj : error NU1101: Unable to find package Humanizer.Core.pt-BR. No packages exist with this id in source(s): darc-pub-dotnet-dotnet-fad253f, dotnet-public, dotnet-tools, dotnet10, dotnet10-transport, dotnet6, dotnet7, dotnet8, dotnet8-transport, dotnet8-workloads, dotnet9, dotnet9-transport, Package source 1, Package source 2, test-tools
/datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj : error NU1101: Unable to find package Humanizer.Core.th. No packages exist with this id in source(s): darc-pub-dotnet-dotnet-fad253f, dotnet-public, dotnet-tools, dotnet10, dotnet10-transport, dotnet6, dotnet7, dotnet8, dotnet8-transport, dotnet8-workloads, dotnet9, dotnet9-transport, Package source 1, Package source 2, test-tools
Failed to restore /datadisks/disk1/work/AE91097D/t/dotnetSdkTests.JSkrMId3/FileBasedApp_---36A7B5DD_5/Program.csproj (in 1.07 sec).
StdErr:


, but found 1 (difference of 1)
```

--- a/test/dotnet.Tests/CommandTests/Package/Add/GivenDotnetPackageAdd.cs
+++ b/test/dotnet.Tests/CommandTests/Package/Add/GivenDotnetPackageAdd.cs
@@ -671,12 +671,19 @@ namespace Microsoft.DotNet.Cli.Package.Add.Tests
         [Theory, CombinatorialData]
         public void FileBasedApp_CentralPackageManagement_NoVersionSpecified_KeepExisting(bool legacyForm, bool fileOption, bool noRestore)
         {
-            if (GetFileBasedAppArgs(legacyForm, versionOption: null, fileOption, noRestore) is not { } args) return;
+            if (GetFileBasedAppArgs(legacyForm, versionOption: null, fileOption, noRestore, packageName: "A") is not { } args) return;
 
             var testInstance = _testAssetsManager.CreateTestDirectory();
+
+            string[] versions = ["0.0.5", "0.9.0", "1.0.0-preview.3"];
+            var packages = versions.Select(e => GetPackagePath(ToolsetInfo.CurrentTargetFramework, "A", e, identifier: e + versions.GetHashCode().ToString())).ToArray();
+
+            var restoreSources = string.Join(";", packages.Select(package => Path.GetDirectoryName(package)));
+
             var file = Path.Join(testInstance.Path, "Program.cs");
-            var source = """
-                #:package Humanizer
+            var source = $"""
+                #:property RestoreSources=$(RestoreSources);{restoreSources}
+                #:package A
                 Console.WriteLine();
                 """;
             File.WriteAllText(file, source);
@@ -688,7 +695,7 @@ namespace Microsoft.DotNet.Cli.Package.Add.Tests
                     <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
                   </PropertyGroup>
                   <ItemGroup>
-                    <PackageVersion Include="Humanizer" Version="2.9.9" />
+                    <PackageVersion Include="A" Version="0.0.5" />
                   </ItemGroup>
                 </Project>
                 """;

